### PR TITLE
Fixed handling legacy DIDs

### PR DESCRIPTION
**Pull request by lukachi**

### CODE REVIEW

File: packages/snap/src/index.ts [hunk 1]
```diff
@@ -155,17 +156,15 @@ export const onRpcRequest = async ({
...
-              identityIdBigIntString: identityStorage.didBigInt,
-            };
-          }
-        } catch (error) {
-          /* empty */
-        }
+      if (
+        identityStorage?.did &&
+        identityStorage?.didBigInt &&
+        isDidSupported(identityStorage.did)
```
Comment (ihordiachenko) [inline packages/snap/src/index.ts:162]: What happens when it's not the supported type?

Comment (lukachi) [reply packages/snap/src/index.ts:162]: creating new identity

Comment (ardier16) [reply packages/snap/src/index.ts:162]: @ihordiachenko @lukachi shouldn't we migrate to new DID type in the background instead?

Comment (lukachi) [reply packages/snap/src/index.ts:162]: done

File: packages/snap/snap.manifest.json [hunk 2]
```diff
@@ -7,21 +7,15 @@
...
+    "shasum": "aH006QfsE9qE68y+zgeP60N3Gc3HB/XJ0AItPeG22cU=",
     "location": {
       "npm": {
         "filePath": "dist/bundle.js",
         "iconPath": "images/icon.svg",
         "packageName": "@rarimo/rarime",
         "registry": "https://registry.npmjs.org/"
       }
-    },
-    "files": [
```
Comment (ihordiachenko) [inline packages/snap/snap.manifest.json]: Why?

File: CHANGELOG.md [hunk 3]
```diff
@@ -10,6 +10,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `@rarimo/rarime`:
   - Creating proof "approve message"
 
+### Fixed
+- `@rarimo/rarime`:
+  - Creating Identity
```
Comment (ihordiachenko) [inline CHANGELOG.md]: Add a description what actually was fixed

File: packages/snap/src/index.ts [hunk 4]
```diff
@@ -155,31 +156,31 @@ export const onRpcRequest = async ({
...
+        Boolean(identityStorage?.did && identityStorage?.didBigInt) ||
+        (await snap.request({
+          method: 'snap_dialog',
+          params: {
+            type: 'confirmation',
+            content: panel([
+              heading('Identity creation'),
+              divider(),
+              text(`You don't have an identity yet`),
+              text('Would you like to create?'),
```
Comment (ihordiachenko) [inline packages/snap/src/index.ts]:
```suggestion
              text('Would you like to create one?'),
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -10,6 +10,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `@rarimo/rarime`:
   - Creating proof "approve message"
 
+### Fixed
+- `@rarimo/rarime`:
+  - `Creating Identity` - added more checks within DID.parse to determine unsupported DID
```
Comment (ihordiachenko) [inline CHANGELOG.md]:
```suggestion
  - `Creating Identity` - added more checks within DID.parse to handle legacy DIDs
```

**Review by ihordiachenko — APPROVED** [review]

**Pull request merged**